### ISSUE TITLE
The experimental Flip Fluid crashes Blender at "baking in progress" stage. 

### ISSUE THREAD

**Issue by Lewiatan45** [rlguy/Blender-FLIP-Fluids#449]
#### System Information

Blender 2.79c
Flip Fluid version 9.0.5.5 (May 17 2018)  
Windows 10 64 bit
I7 6700k 4.40 Ghz
MSI Nvidia Gefore 1080 8GB
48 GB DDR4 2660 Mhz 

#### Describe the bug

After the add-on finishes exporting animated mesh, it crashes blender at "baking in progress". 

#### To Reproduce

I selected time frame. Pressed bake, addon finishes exporting the animated mesh and then it crashes Blender

#### Expected Behaviour

The add-on should be able to bake fluids regardless of length and complexity of animation.

#### Actual Behaviour

As it was described above. I selected time frame. Pressed bake, addon finishes exporting the animated mesh and then it crashes Blender. My animation is 2700 frame long and fluids should appear at the 1695th frame and stay till the end of the animation. There are 7 fluids sources. In the previous stable version of this add-on, the Blender didn't crash if I told add-on to bake short part of the animation for example 30-60 frames only.

#### Screenshots
Flip Fluid didn't create any log files in the fluid cache folder, but I managed to get the name of the error. It seems there is a problem with the allocation of memory.

[Bug Flip Fluid.txt](https://github.com/rlguy/Blender-FLIP-Fluids/files/3193400/Bug.Flip.Fluid.txt)

**Comment by rlguy** [comment]
Hi Lewiatan45,

Thanks for the report! Would it be possible for you to attach the .blend file so that we can test this issue? You may attach it here or send it to us at flip.fluids@gmail.com.

The **std::bad_alloc** error can be caused by the Operating System not being able to allocate a chunk of memory. This could be caused by a piece a memory that is too large, or by a bug that causes us to allocate a negative amount of memory.

A few questions that could help narrow down this issue:
- What are the simulation 3d grid dimensions? This can be found by Enabling the 'Display Grid' option in the Debug panel.
- Does the error still occur if you reduce the frame length as you did in the earlier version?
- Does the error still occur if you delete the animated meshes?
- Does the error still occur if you delete the fluid sources?
- Does the Blender memory usage value rise to a very large amount during mesh export?

![memory_value](https://user-images.githubusercontent.com/4285191/57961614-03411c80-78c5-11e9-844b-3bc373d538a6.jpg)

Note: 590K triangles for an animated mesh is quite a lot of data and much more detail than what would be needed in fluid simulation. Do you have any subdivisions to the character that you are able to reduce to generate a smaller mesh? High density animated meshes like this could cause performance issues during simulation, especially if the polys are much smaller than a grid cell. 

\- Ryan

**Comment by Lewiatan45** [comment]
No, I can't send you blend file.

- What are the simulation 3d grid dimensions? This can be found by Enabling the 'Display Grid' option in the Debug panel.

![Display Grid](https://user-images.githubusercontent.com/26897892/57969613-3cd35f80-7979-11e9-9538-da7981a29a96.JPG)

- Does the error still occur if you reduce the frame length as you did in the earlier version?
Yes, it does.

- Does the error still occur if you reduce the simulation resolution? Such as half the value of the current setup.
Yes, it does.

- Does the error still occur if you delete the animated meshes?
Yes, it does.

- Does the error still occur if you delete the fluid sources?
Yes, it does.

-Does the Blender memory usage value rise to a very large amount during mesh export?
It increases frin 852 MB to 880MB.

The first model has 36k polygons, the second one has 37k polygons, the last one 4k polygons(terrain)
The scene is a forest with a lot of trees and vegetation. 

I managed to find work around this issue. I removed ever mesh from the scene that was not involved in the baking process. Then Flip Fluid managed to bake the fluid without crashing Blender. The resolution and the length of animation remained the same. 200 resolution and length from 1656 to 2700 frames. It seems like add-on "read" every mesh that is in the scene instead of ignoring meshes that are not involved in the baking process. Therefore it crashed Blender at "baking in progress". Maybe that's why you saw 590k triangles for animated meshes in the error log I sent you.

**Comment by rlguy** [comment]
Hi Lewiatan45,

Thanks for testing this out and for the update!

I was not able to reproduce the addon exporting meshes not related to the simulation. From the error log, it looks like the following objects are being exported:

### Static Meshes:

| Object | Verts | Triangles | Filesize |
|---|---|---|---|
| CF.Back.L |   386 |   768 |   14 kB |
| CF.Front.L |   386 |   768 |   14 kB |
| Sphere |   482 |   960 |   17 kB |
| CF.Front.L.001 |   386 |   768 |   14 kB |
| Ziemia |   2025 |   3872 |   69 kB |
| CF.Front |   386 |   768 |   14 kB |
| CF.Front.R.001 |   386 |   768 |   14 kB |
| CF.Back.R |   386 |   768 |   14 kB |

### Animated Meshes:

| Object | Verts | Triangles | Filesize |
|---|---|---|---|
| Body.D |   32451 |   64886 |   1.1 MB |
| Body |   295268 |   588144 |   10.1 MB |

- Is the Body object involved with the simulation? It will be exported for the simulator if it is set to a FLIP Fluid object type or if it is set as an inflow velocity target.
- Does the Body object have any modifiers that could increase the poly count of the object?
- Does the Body object have any particle systems? Blender evaluates particle systems when converting an object to a mesh.
- Is the Body object large compared to the Domain? The simulator must make some calculations over the entire object and If the object is larger than the domain, the simulator will wrap a larger domain around the object so that it can process the entire object volume. If so, this could explain the **std::bad_alloc** (out of memory) error.

\- Ryan

**Comment by Lewiatan45** [comment]
- Is the Body object involved with the simulation? It will be exported for the simulator if it is set to a FLIP Fluid object type or if it is set as an inflow velocity target.
Yes, it is. Body.D and Body objects have FLIP Fluids physics set to obstacles. 

-Does the Body object have any modifiers that could increase the poly count of the object?
No, it doesn't.

-Does the Body object have any particle systems? Blender evaluates particle systems when converting an object to a mesh.
No, it doesn't.

-Is the Body object large compared to the Domain? The simulator must make some calculations over the entire object and If the object is larger than the domain, the simulator will wrap a larger domain around the object so that it can process the entire object volume. If so, this could explain the std::bad_alloc (out of memory) error.
Yes, the Body object is large comparing to the Domain. Your add-on is designed in the way that users need to balance resolution, the size of the domain or the size of the inflow object because if one of them isn't in the right size, the fluid will not be generated at all. So if I had a large domain and really small inflow object, the fluid wouldn't be generated. In such a case, I'd have to increase the resolution to a ridiculous number to get fluids with such setup. That's why I keep the domain small comparing to the model. 

All in all, I removed all objects that do not have FLIP Fluids physics turned on and then I was able to bake fluids. With those objects present in the scene, Blender was crashing at "baking in progress" even though they were not involved in the baking process.
